nexus/src/nexus.rs

```diff
@@ -574,8 +581,9 @@ impl Nexus {
     ) -> LookupResult<db::model::Project> {
         let organization_id = self
             .db_datastore
-            .organization_lookup_id_by_name(organization_name)
-            .await?;
+            .organization_lookup_id(organization_name)
```

Comment:
In this file, there were a whole bunch of callers of `organization_lookup_id_by_name(name)`, which returned a plain Uuid for the Organization.  Rather than keep that confusing extra function, since the transformation is so easy, I just updated all the callers to use `organization_lookup_id(name)` instead.

Most (all?) of these callers are doing this lookup in order to look up something else inside the Organization.  We will eventually want to change those lookup functions to accept an `authz::Organization` rather than an Organization uuid so that when you finally reach the thing you're looking for (say, an `authz::Project`), you can do the appropriate access check.  I'm trying to do this incrementally, so for now, I've not changed the behavior of these call sites.